### PR TITLE
openapi_tool

### PR DESCRIPTION
**Pull request by wangyijunlyy**
Refer to test_openapi_tool.py,we need to run openapi_schema_convert() function firstly,writed in the openapi_tool.py.
Then we can call the class OpenAPISchemaTool() in the openapi_tool.py to get remote call response.

### CODE REVIEW

**Comment by wangyijunlyy** [comment]
test_token='sk-f4b777268d394aa68ad9800511b6ab5a'

File: modelscope_agent/tools/openapi_tool.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
+import sys
+sys.path.append("../../")
```
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 主库里面去掉这个

File: modelscope_agent/tools/openapi_tool.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    description: str
+    required: Optional[bool] = True
+
+
+class ToolSchema(BaseModel):
+    name: str
+    description: str
+    parameters: List[ParametersSchema]
+
+class OpenAPISchemaTool():
```
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 继承基础Tool

File: modelscope_agent/tools/openapi_tool.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+                'description': self.description,
+                'parameters': self.parameters
+            }
+            self.tool_schema = ToolSchema(**all_para)
+        except ValidationError:
+            raise ValueError(f'Error when parsing parameters of {self.name}')
+        self._str = self.tool_schema.model_dump_json()
+        self._function = self.parse_pydantic_model_to_openai_function(all_para)
+
+    def __call__(self, remote=False, *args, **kwargs):
```
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 继承基础tool后，这个方法可以不用了

File: modelscope_agent/tools/openapi_tool.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+                    raise ValueError(
+                        f'Remote call failed with error code: {e.response.status_code},\
+                        error message: {e.response.content.decode("utf-8")}')
+
+            raise ValueError(
+                'Remote call max retry times exceeded! Please try to use local call.'
+            )
+        else:
+            raise ValueError('Remote call method is invalid!We have POST and GET method.')
+    def _local_call(self, *args, **kwargs):
```
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 删掉这个方法

File: modelscope_agent/tools/openapi_tool.py [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+        return 
+    
+    def _remote_parse_input(self,*args, **kwargs):
+        model_name = kwargs.pop('model', '')
+        if model_name == '':
+          model_name = self.parameters.get('value')[0]
+        kwargs['model'] = model_name
+        restored_dict = {}
+        for key, value in kwargs.items():
+            # 检查键中是否包含 "."
```
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 中文换成英文

File: modelscope_agent/tools/openapi_tool.py (hunk 4 above)
Comment (zzhangpurdue) [inline modelscope_agent/tools/openapi_tool.py]: 格式不对，注意用pre-commit install & pre-commit run --all-files 过一下lint格式检查

**Pull request closed without merging**